cpp/test/sg/dbscan_test.cu

```diff
@@ -203,7 +204,7 @@ class Dbscan2DSimple : public ::testing::TestWithParam<DBScan2DArrayInputs<T>> {
     CUDA_CHECK(cudaStreamSynchronize(handle.get_stream()));
 
     Dbscan::fit(handle, inputs, (int)params.n_row, 2, params.eps,
-                params.min_pts, labels, core_sample_indices_d);
+                params.min_pts, Dbscan::L2, labels, core_sample_indices_d);
```

Comment:
Consider making the metric type a field of DBScan2DArrayInputs, and add a small test example with a precomputed distances.